### ISSUE TITLE
Unstructured response errors from `doInitialize`

### ISSUE THREAD

**Issue by samcowger** [haskell/lsp#586]
As of lsp-2.6.0.0/lsp-types-2.2.0.0, the `doInitialize` field of `ServerDefinition` is a function that evaluates to an `IO (Either ResponseError a)`. If I'm not mistaken, the `ResponseError` component of that type could/should be a `TResponseError 'Method_Initialize`. 

This would only change the type of the `_xdata` portion of the error, refining it from its current type of `Aeson.Value` down to `InitializeError`, which would be in line with the [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initializeResult).

**Comment by michaelpj** [comment]
Yes, this is, sadly, deliberate. I think the reason was that it turned out that HLS was passing around untyped `ResponseError`s a lot, and it was going to be a pain at the time to convert them. Obviously that's doing things backwards, and we should just adapt HLS, but I haven't got to it yet. I would definitely like to do this!
